flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/deserializers/SdkBindingDataDeserializer.java

```diff
@@ -0,0 +1,155 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.flyte.flytekit.jackson.deserializers;
+
+import static java.util.stream.Collectors.toList;
+import static org.flyte.flytekit.jackson.serializers.SdkBindingDataSerializationProtocol.KIND;
+import static org.flyte.flytekit.jackson.serializers.SdkBindingDataSerializationProtocol.LITERAL;
+import static org.flyte.flytekit.jackson.serializers.SdkBindingDataSerializationProtocol.SCALAR;
+import static org.flyte.flytekit.jackson.serializers.SdkBindingDataSerializationProtocol.TYPE;
+import static org.flyte.flytekit.jackson.serializers.SdkBindingDataSerializationProtocol.VALUE;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.flyte.api.v1.Literal;
+import org.flyte.api.v1.LiteralType;
+import org.flyte.api.v1.Primitive;
+import org.flyte.api.v1.Scalar;
+import org.flyte.flytekit.SdkBindingData;
+
+class SdkBindingDataDeserializer extends StdDeserializer<SdkBindingData<?>> {
+  private static final long serialVersionUID = 0L;
+
+  public SdkBindingDataDeserializer() {
+    super(SdkBindingData.class);
+  }
+
+  private SdkBindingData<?> transform(JsonNode tree) {
+    Literal.Kind literalKind = Literal.Kind.valueOf(tree.get(LITERAL).asText());
+    switch (literalKind) {
+      case SCALAR:
+        switch (Scalar.Kind.valueOf(tree.get(SCALAR).asText())) {
+          case PRIMITIVE:
+            switch (Primitive.Kind.valueOf(tree.get("primitive").asText())) {
+              case INTEGER_VALUE:
+                return SdkBindingData.ofInteger(tree.get(VALUE).longValue());
+              case BOOLEAN_VALUE:
+                return SdkBindingData.ofBoolean(tree.get(VALUE).booleanValue());
+              case STRING_VALUE:
+                return SdkBindingData.ofString(tree.get(VALUE).asText());
+              case DURATION:
+                return SdkBindingData.ofDuration(Duration.parse(tree.get(VALUE).asText()));
+              case DATETIME:
+                return SdkBindingData.ofDatetime(Instant.parse(tree.get(VALUE).asText()));
+              case FLOAT_VALUE:
+                return SdkBindingData.ofFloat(tree.get(VALUE).doubleValue());
+            }
+            break;
+          case GENERIC:
+          case BLOB:
+            // TODO: We need to implement this
+            throw new RuntimeException("not supported");
```

Comment:
Give better error message and unify the specific exception, for example generic RuntimeException vs IllegalArgumentException vs UnsupportedOperation Exception

Reply:
Which type of exception do you suggest that we use?

Reply:
We chatted and decided on `IllegalArgumentException` because the serialization and deserialization are executed because the user did `JacksonSdkType.of(Input.class)`.
```suggestion
            throw new IllegalArgumentException("Type contains an unsupported scalar: " + scalarKind);
```
assuming that we wrote on line 58 
```java
Scalar.Kind scalarKind = Scalar.Kind.valueOf(tree.get(SCALAR).asText());
switch (scalarKind) {
```